pkg/templates/cache.go

```diff
@@ -1,18 +1,26 @@
 package templates
 
 import (
+	"sync"
+
 	"github.com/projectdiscovery/utils/conversion"
 	mapsutil "github.com/projectdiscovery/utils/maps"
 )
 
 // Templates is a cache for caching and storing templates for reuse.
 type Cache struct {
 	items *mapsutil.SyncLockMap[string, parsedTemplate]
+
+	// parsedTemplatePool reduces allocation overhead when storing cache entries
+	parsedTemplatePool *sync.Pool
```

Comment:
do we really need this? Ig storing items in mapsutil cache is enoug right as anyway we aren't caching compiled templates?